parser/src/main/kotlin/tools/samt/parser/Parser.kt

```diff
@@ -533,15 +593,38 @@ class Parser private constructor(
         val expectedString = getHumanReadableName<T>()
         val gotString = current!!.getHumanReadableName()
 
-        if (current is EndOfFileToken) {
-            reportFatalError("Expected '${expectedString}' but reached end of file")
+        if (isEnd) {
+            diagnostic.fatal {
+                message("Expected '${expectedString}' but reached end of file")
+                highlight(current!!.location, highlightBeginningOnly = true)
+            }
         }
 
-        if (T::class == IdentifierToken::class && current is StaticToken) {
-            reportFatalError("'${gotString}' is a reserved keyword, did you mean to escape it? (e.g. '^${gotString}')")
+        if (T::class == IdentifierToken::class && check<StaticToken>()) {
+            diagnostic.error {
+                message("Unescaped identifier '${gotString}'")
+                highlight(current!!.location, suggestChange = "^${gotString}")
+                help("To use '${gotString}' as an identifier, escape it with a caret: '^${gotString}'")
+            }
+
+            // we can continue parsing the file treating the static token as an identifier
+            val staticToken = expectOrNull<StaticToken>()
+            require(staticToken != null)
```

Comment:
Could be `requireNotNull`
```suggestion
            requireNotNull(staticToken)
```